auto_nag/scripts/assignee_no_login.py

```diff
@@ -46,22 +44,30 @@ def handle_bug(self, bug, data):
         if self.people.is_mozilla(assignee):
             return None
 
-        bugid = str(bug["id"])
-        data[bugid] = {"triage_owner": bug["triage_owner_detail"]["real_name"]}
-        prod = bug["product"]
-        comp = bug["component"]
-        default_assignee = self.default_assignees[prod][comp]
-        self.autofix_assignee[bugid] = {"assigned_to": default_assignee}
-        return bug
-
-    def get_mail_to_auto_ni(self, bug):
         # Avoid to ni everyday...
         if self.has_bot_set_ni(bug):
             return None
 
-        mail = bug["triage_owner"]
-        nick = bug["triage_owner_detail"]["nick"]
-        return {"mail": mail, "nickname": nick}
+        bugid = str(bug["id"])
+
+        if bug["triage_owner"] in self.auto_needinfo:
+            info = self.auto_needinfo[bug["triage_owner"]]
+            if len(info["bugids"]) >= self.max_ni:
+                # Don't unassign if we can't needinfo.
+                return None
+
+            info["bugids"].append(str(bugid))
+        else:
+            self.auto_needinfo[bug["triage_owner"]] = {
+                "nickname": bug["triage_owner_detail"]["nick"],
+                "bugids": [bugid],
+            }
```

Comment:
It may be cleaner if this logic moved in a method that override `add_auto_ni()` then use `add_auto_ni()` here. 

Another idea: maybe it is worth implementing this logic in `BzCleaner`, it is always good to not overwhelm people with tons of _ni_ at once.

If we go with any of these routes, we should modify `add_auto_ni()` to expect to return an indicator if the _ni_ is added.

Reply:
We have some logic in BzCleaner for a max needinfo number, but it doesn't help in this case (if we use the built-in logic, when we exceed the number of needinfos we start unassigning people without needinfoing).

Reply:
In the case of this tool, if the method returns an indicator of whether the _ni_ got added or not, we can use that to control the following steps (e.g., unassigning people). 

But I agree with you; this could cause unexpected behaviour to other tools.  We need to check how other tools use the method before generalizing this behaviour (i.e., 5 other tools are using `add_auto_ni()`). 

A safe option is to override `add_auto_ni()` for now. We can follow up to generails the behaviour in a separate PR.

Reply:
Filed #1383 as a follow-up.